### DISCUSSION TITLE
'Diligent::BufferVkImpl::CreateView' hides overloaded virtual function

### DISCUSSION THREAD

**Discussion by HonnyChen** [DiligentGraphics/DiligentEngine#232] [Q&A]
I try to build my project with Diligent on MacOS. And I got some errors. "hides overloaded virtual function". How can I do to clear these errors?

**Comment by TheMostDiligent** [comment]
You added a function that is named the same as a virtual function in the bare class

**Reply by HonnyChen** [reply]
It's really could pass the MacOS build on v2.5.3? I think this issue can't pass MacOS build. Don't you have this issue on MacOS build?
![圖片](https://user-images.githubusercontent.com/5293331/218949258-1c8165c7-5ce5-45f8-b448-efb05fb4b973.png)

**Reply by TheMostDiligent** [reply]
> It's really could pass the MacOS build on v2.5.3?

Yes. This is [the last build on CI](https://github.com/DiligentGraphics/DiligentEngine/actions/runs/4171820688).

Is this Diligent source code without modifications?

**Reply by HonnyChen** [reply]
Yes, It's original Diligent source code without modifications. I integrate Diligent as a third lib into my project without modifications.
XCode is 14.0.1 and MacOS is 12.6.

**Reply by TheMostDiligent** [reply]
Do you use Diligent's CMake?

**Reply by HonnyChen** [reply]
Yes, I use Diligent's CMake via add_subdirectory under my cmake project.

**Reply by TheMostDiligent** [reply]
Maybe there are some updates to xcode that broke it again - this happens with every single xcode update. I will take a look
